File: frontend/devicetrackr-web/src/app/models/device.ts
import { User } from './user';

export interface Device {
  id: number;
  name: string;
  manufacturer: string;
  type: number;
  operatingSystem: string;
  osVersion: string;
  processor: string;
  ramAmountGb: number;
  description: string;
  assignedUserId?: number | null;
  assignedUser?: User | null;
}

export interface DevicePayload {
  name: string;
  manufacturer: string;
  type: number;
  operatingSystem: string;
  osVersion: string;
  processor: string;
  ramAmountGb: number;
  description: string;
}

File: backend/DeviceTrackr.Api/Models/Device.cs
namespace DeviceTrackr.Api.Models;

public class Device
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Manufacturer { get; set; } = string.Empty;
    public DeviceType Type { get; set; }
    public string OperatingSystem { get; set; } = string.Empty;
    public string OsVersion { get; set; } = string.Empty;
    public string Processor { get; set; } = string.Empty;
    public int RamAmountGb { get; set; }
    public string Description { get; set; } = string.Empty;
    public int? AssignedUserId { get; set; }
    public User? AssignedUser { get; set; }
}
